src/parser/SGParser.ts

```diff
@@ -0,0 +1,353 @@
+import type { AttributeCstNode, ContentCstNode, DocumentCstNode, ElementCstNode } from '@xml-tools/parser';
+import * as parser from '@xml-tools/parser';
+import { DiagnosticMessages } from '../DiagnosticMessages';
+import type { Diagnostic, Range } from 'vscode-languageserver';
+import util from '../util';
+import { SGProlog, SGChildren, SGComponent, SGField, SGFunction, SGInterface, SGNode, SGScript } from './SGTypes';
+import type { SGAst, SGTag, SGToken, SGAttribute, SGReferences } from './SGTypes';
+import { isSGComponent } from '../astUtils/xml';
+
+export default class SGParser {
+
+    /**
+     * The AST of the XML document, not including the inline scripts
+     */
+    public ast: SGAst = {};
```

Comment:
Perhaps this should be a class, just like the other SG AST elements. Then the vast majority of [XmlFile#transpileAst](https://github.com/rokucommunity/brighterscript/pull/235/files#diff-c2462ef143ff214d1683501300a6cc2960cf54d7ba36e7606f7383c95021a240R486) can be moved into this new class. 

I did something similar with the `Body` statement in the BrighterScript parser.